src/pathOrderOptimizer.cpp

```diff
@@ -126,13 +90,77 @@ int PathOrderOptimizer::getClosestPointInPolygon(Point prev_point, int poly_idx)
     {
         const Point& p1 = poly[point_idx];
         const Point& p2 = poly[(point_idx + 1) % poly.size()];
-        int64_t dist = vSize2(p1 - prev_point);
-        float is_on_inside_corner_score = -LinearAlg2D::getAngleLeft(p0, p1, p2) / M_PI * 5000 * 5000; // prefer inside corners
-        // this score is in the order of 5 mm
-        if (dist + is_on_inside_corner_score < best_point_score)
+        // when type is SHARPEST_CORNER, actual distance is ignored, we use a fixed distance and decision is based on curvature only
+        int64_t dist_score = (config.type == EZSeamType::SHARPEST_CORNER)? 100 : vSize2(p1 - prev_point);
+        const float corner_angle = LinearAlg2D::getAngleLeft(p0, p1, p2) / M_PI; // 0 -> 2
+        if (config.type == EZSeamType::SHORTEST)
```

Comment:
There's a lot of code duplication between this if branch and the else branch.
You can simplify it a lot by combining the two branches and merging the corner_shift with the dist/scaling_factor:
```
const int64_t corner_shift = (type == EZSeamType::SHORTEST)? 10000 * 10000 : dist / 50;
```

Reply:
I think I would rather leave the duplicated code until we are happy that the two strategies are working OK.

Reply:
The way we work is that first we review the code of a PR and then we test whether it works alright. According to our workflow I cannot move on to testing until the code is okay.
You could also remove the code duplication like this:
```
float corrected_corner_angle;
(config.corner_pref)
{
  case EZSeamCornerPrefType::Z_SEAM_CORNER_PREF_INNER:
    corrected_corner_angle = std::max(0, corner_angle - 1);
...
}
if (config.type == EZSeamType::SHORTEST)
{
  dist -= corrected_corner_angle * corner_shift;
}
else
{  dist -= corrected_corner_angle * dist_score / scaling_divisor; }
```